pygeo/parameterization/BaseDVGeo.py

```diff
@@ -0,0 +1,223 @@
+            Dictionary of design variables. The keys of the dictionary must correspond to the design variable names.
+            Any additional keys in the dictionary are simply ignored.
+        """
+        pass
+
+    @abstractmethod
+    def totalSensitivity(self, dIdpt, ptSetName, comm=None):
+        r"""
+        This function computes sensitivity information.
+
+        Specifically, it computes the following:
+        :math:`\frac{dX_{pt}}{dX_{DV}}^T \frac{dI}{d_{pt}}`
+
+        Parameters
+        ----------
+        dIdpt : array of size (Npt, 3) or (N, Npt, 3)
+            This is the total derivative of the objective or function of interest with respect to the coordinates in 'ptSetName'.
+            This can be a single array of size (Npt, 3) **or** a group of N vectors of size (Npt, 3, N).
+            If you have many to do, it is faster to do many at once.
+
+        ptSetName : str
+            The name of set of points we are dealing with
+
+        comm : MPI.IntraComm
+            The communicator to use to reduce the final derivative. If comm is None, no reduction takes place.
+
+        Returns
+        -------
+        dIdxDict : dic
+            The dictionary containing the derivatives, suitable for pyOptSparse
+        """
+        pass
+
+    @abstractmethod
+    def totalSensitivityProd(self, vec, ptSetName):
+        r"""
+        This function computes sensitivity information.
+
+        Specifically, it computes the following:
+        :math:`\frac{dX_{pt}}{dX_{DV}} \times\mathrm{vec}`
+
+        This is useful for forward AD mode.
+
+        Parameters
+        ----------
+        vec : dictionary whose keys are the design variable names
+            and whose values are the derivative seeds of the corresponding design variable.
+
+        ptSetName : str
+            The name of set of points we are dealing with
+
+        comm : MPI.IntraComm
+            The communicator to use to reduce the final derivative. If comm is None, no reduction takes place.
+
+        Returns
+        -------
+        xsdot : array (Nx3) -> Array with derivative seeds of the surface nodes.
+        """
+        pass
+
+    @abstractmethod
+    def update(self, ptSetName):
+        """
+        This is the main routine for returning coordinates that have been updated by design variables.
+
+        Parameters
+        ----------
+        ptSetName : str
+            Name of point-set to return. This must match ones of the given in an :func:`addPointSet()` call.
+        """
+        pass
+
+    @abstractmethod
+    def writePointSet(self, name, fileName):
+        """
+        Write a given point set to a tecplot file
+
+        Parameters
+        ----------
+        name : str
+             The name of the point set to write to a file
+
+        fileName : str
+           Filename for tecplot file. Should have no extension, an extension will be added
+        """
+        coords = self.update(name, childDelta=True)
```

Comment:
Since the regular DVGeo has to have the child parameters I'll keep its writePointSet in its original state. I'll move this version (minus the childDelta) to the middle class for ESP and VSP to use and keep the top level class one as an abstract method.